stagemonitor-web-servlet/src/main/java/org/stagemonitor/web/servlet/eum/ClientSpanTagProcessor.java

```diff
@@ -29,24 +31,23 @@ protected ClientSpanTagProcessor(String weaselOperationTypeToProcess, Collection
 		this.requiredParams = requiredParams;
 	}
 
-	public void processSpanBuilder(Tracer.SpanBuilder spanBuilder, Map<String, String[]> servletRequestParameters) {
-		if (shouldProcess(servletRequestParameters)) {
-			processSpanBuilderImpl(spanBuilder, servletRequestParameters);
+	public void processSpanBuilder(Tracer.SpanBuilder spanBuilder, Map<String, String[]> requestParameters) {
+		if (shouldProcess(requestParameters)) {
+			processSpanBuilderImpl(spanBuilder, requestParameters);
 		}
 	}
 
-	protected void processSpanBuilderImpl(Tracer.SpanBuilder spanBuilder, Map<String, String[]> servletRequestParameters) {
+	protected void processSpanBuilderImpl(Tracer.SpanBuilder spanBuilder, Map<String, String[]> requestParameters) {
 		// default no-op
 	}
 
-	protected boolean shouldProcess(Map<String, String[]> servletRequestParameters) {
+	protected boolean shouldProcess(Map<String, String[]> requestParams) {
 		for (String requiredParam : requiredParams) {
-			if (StringUtils.isEmpty(getParameterValueOrNull(requiredParam, servletRequestParameters))) {
+			if (StringUtils.isEmpty(getParameterValueOrNull(requiredParam, requestParams))) {
 				return false;
 			}
 		}
-		final String type = getParameterValueOrNull(ClientSpanServlet.PARAMETER_TYPE, servletRequestParameters);
-		return typeToProcess != null && typeToProcess.equals(type);
+		return typeToProcess == null || typeToProcess.equals(getParameterValueOrNull(PARAMETER_TYPE, requestParams));
```

Comment:
I remember discussing this in the first draft regarding `null` for all operation types -> maybe we should introduce a new `ALL`-constant in `ClientSpanTagProcessor`? It would be IMHO cleaner than a null pseudotype. `ClientSpanTypeTagProcessor` would then also not need to override `shouldProcess` anymore.